src/main/java/org/prebid/server/events/EventUtil.java

```diff
@@ -39,15 +43,27 @@ public class EventUtil {
 
     private static final String LINE_ITEM_ID_PARAMETER = "l";
 
+    private static final Set<String> POSSIBLE_TYPES = stringValues(EventRequest.Type.class);
+    private static final Set<String> POSSIBLE_V_TYPES = stringValues(EventRequest.VastType.class);
+
     private EventUtil() {
     }
 
     public static void validateType(RoutingContext routingContext) {
         final String type = routingContext.request().params().get(TYPE_PARAMETER);
-        if (ObjectUtils.notEqual(type, IMP_TYPE) && ObjectUtils.notEqual(type, WIN_TYPE)) {
+        if (!POSSIBLE_TYPES.contains(type)) {
             throw new IllegalArgumentException(String.format(
-                    "Type '%s' is required query parameter. Possible values are %s and %s, but was %s",
-                    TYPE_PARAMETER, WIN_TYPE, IMP_TYPE, type));
+                    "Type '%s' is required query parameter. Possible values are %s, but was %s",
+                    TYPE_PARAMETER, POSSIBLE_TYPES, type));
+        }
+    }
+
+    public static void validateVType(RoutingContext routingContext) {
+        final String vType = routingContext.request().params().get(V_TYPE_PARAMETER);
+        if (!POSSIBLE_V_TYPES.contains(vType)) {
+            throw new IllegalArgumentException(String.format(
+                    "Type '%s' is required query parameter. Possible values are %s, but was %s",
```

Comment:
`vtype` parameter is not always required. As you can see [here](https://github.com/prebid/prebid-server/pull/2179/files#diff-ed708c8cdfaf7ce0ada46a2901228d5a2ed5d6d9bc59981b3eda6ee02111bd19R148) it is required iff `eventType == VAST`.

Reply:
Yeah... you're right, sorry missed this case